javagen/src/main/java/com/azure/autorest/template/ClientMethodTemplateBase.java

```diff
@@ -111,7 +111,8 @@ private static void optionalParametersJavadoc(String title, List<ProxyMethodPara
         commentBlock.line(String.format("    <caption>%s</caption>", title));
         commentBlock.line("    <tr><th>Name</th><th>Type</th><th>Required</th><th>Description</th></tr>");
         for (ProxyMethodParameter parameter : parameters) {
-            if (!parameter.getIsRequired()) {
+            boolean parameterIsConstantOrFromClient = parameter.getIsConstant() || parameter.getFromClient();
+            if (!parameter.getIsRequired() && !parameterIsConstantOrFromClient) {
```

Comment:
do you need to modify `hasOptionalParameters` as well? And maybe rename it (as it no longer for optional parameter only.